src/Sylius/Bundle/PayumBundle/Validator/GatewayConfigurationValidator.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\PayumBundle\Validator;
+
+use Sylius\Bundle\PayumBundle\Validator\Constraints\GatewayConfig;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+final class GatewayConfigurationValidator extends ConstraintValidator
+{
+    /** @var array<string, mixed> */
+    private array $factoriesValidators;
+
+    /**
+     * @param array<string, string> $factoryNames
+     * @param array<string, mixed> $factoriesValidators
+     */
+    public function __construct(private array $factoryNames, iterable $factoriesValidators)
+    {
+        $this->factoriesValidators = $factoriesValidators instanceof \Traversable ? iterator_to_array($factoriesValidators) : $factoriesValidators;
+    }
+
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        /** @var GatewayConfig $constraint */
+        Assert::isInstanceOf($constraint, GatewayConfig::class);
+
+        if (!in_array($value->getFactoryName(), array_keys($this->factoryNames), true) || $value->getFactoryName() === 'offline') {
+            return;
```

Comment:
IMO, we should add validation here that the factoryName is wrong

Reply:
what about when someone uses different gateway?